File: livingdoc-fixture-converter/src/main/java/org/livingdoc/fixture/converter/time/OffsetDateTimeConverter.java
package org.livingdoc.fixture.converter.time;

import java.time.OffsetDateTime;
import java.time.format.DateTimeFormatter;


public class OffsetDateTimeConverter extends AbstractTemporalConverter<OffsetDateTime> {

    @Override
    DateTimeFormatter getDefaultFormatter() {
        return DateTimeFormatter.ISO_DATE_TIME;
    }

    @Override
    OffsetDateTime parse(String value, DateTimeFormatter formatter) {
        return OffsetDateTime.parse(value, formatter);
    }

}

File: livingdoc-converters/src/main/kotlin/org/livingdoc/converters/time/OffsetDateTimeConverter.kt
package org.livingdoc.converters.time

import java.time.OffsetDateTime
import java.time.format.DateTimeFormatter
import kotlin.reflect.KClass

/**
 * This converter parses a String to the time format with offset from UTC
 */
open class OffsetDateTimeConverter : AbstractTemporalConverter<OffsetDateTime>() {

    override fun defaultFormatter(): DateTimeFormatter = DateTimeFormatter.ISO_OFFSET_DATE_TIME
    override fun doParse(value: String, formatter: DateTimeFormatter): OffsetDateTime =
        OffsetDateTime.parse(value, formatter)

    override fun canConvertTo(targetType: KClass<*>) = OffsetDateTime::class == targetType
}
